### ISSUE TITLE
Interest in manifold of matrices with orthogonal columns?

### ISSUE THREAD

**Issue by marfiori** [pymanopt/pymanopt#237]
Hi all, I've been working with constraints of orthogonality of the columns of a matrix, but not orthonormality. This is, instead of the Stiefel manifold, this is the manifold of matrices where XX^T is diagonal.
I have an implementation of this manifold with projections and a retraction.
If there's interest in this, I'd be happy to do the PR (I would need help naming the manifold)

**Comment by kellertuer** [comment]
Hi,
this sounds very interesting. Just to clarify, since I do not see the detail here. Is your manifold

$\mathcal M_D = \bigl\\{ X \in R^{n,p}$ such that $X^{\mathrm{T}}X = D\bigr\\}$ 

or

$\mathcal M = \bigl\\{ X \in R^{n,p}$ such that $X^{\mathrm{T}}X = D$ for some $D$ a diagonal $p,p$ matrix $\bigr\\}$ ?

For the first I am not sure what that is, that would be interesting then whether you have literature on that :)

For the second I think that is very close if not the same as the (square) fixed rank matrices, cf. [here](https://pymanopt.org/docs/stable/manifolds.html#module-pymanopt.manifolds.fixed_rank), or [here](https://juliamanifolds.github.io/Manifolds.jl/stable/manifolds/fixedrankmatrices.html) for a few more detailed formulae or [here](https://www.manopt.org/reference/manopt/manifolds/fixedrank/fixedrankembeddedfactory.html).

**Comment by marfiori** [comment]
Hi Ronny, thanks for the answer.

It's the second version (the resulting diagonal matrix is not fixed). Just to be sure, I'll write it in a different way:

$\mathcal{M}(d,n) = \lbrace X \in \mathbb{R}^{n\times d}: X^\top X \text{ is diagonal}\rbrace = \lbrace X \in \mathbb{R}^{n\times d}: W\circ (X^\top X)=0\rbrace$

where $\circ$ is the Hadamard (entry-wise) product, and $W=\mathbf{1}\mathbf{1}^\top-I_d$ is a mask matrix, with zeros on the diagonal and ones everywhere else. 

However, I think it's not the same as the manifold of fixed-rank matrices. I'm interested in the matrices $X$ (like a generalized version of Stiefel), not in the resulting $X^\top X$.

I forked the repository, added this manifold, and I used it to solve some optimization coming from an embedding problem. This is part of a paper we're submitting soon.

**Comment by kellertuer** [comment]
Interesting. I got to that observation due to Vandereycken – or see the links I posted – where he stores the n-by-k rank p matrix as an SVD.
Now if in his setting n=k and his rank p is your d, we get quite close to your representation and he also stores the SVD representation A = USV, where for n=k (swarm matrices) and symmetric ones (that is where its not completely the same) I think he has the same representation as you do.

But you are right, while the representation and such is similar, your interest is different.

Ah! But I saw another connection at least to my first idea, if you check https://juliamanifolds.github.io/Manifolds.jl/latest/manifolds/generalizedstiefel.html and set the B in there to be the diagonal matrix with the inverses of your diagonal. But this is what I first had in mind ;) (and I think only in Manifolds.jl not in pymanopt available).

Notation wise I would prefer more like $X^{\mathrm{T}}X = \operatorname{diag}(d_1,\ldots,d_p)$, $0 \neq d_i \in \mathbb R$, $i=1,\ldots,d$ or so, that is maybe easier than a Hadamard product (to say the rest is 0) – as well as, I _think_ you do not want the diagonal to be zero, which your writing does not yet exclude, at least for Stiefel you always have nonzero diagonals.

**Comment by marfiori** [comment]
Yes, exactly, I don't want zeros in that diagonal. I simplified the notation we use in the paper, but I missed that detail. Thanks.

I think that this GeneralizedStiefel trick could be useful for the first example you mentioned in your first post, where the matrix D is fixed (which also has interesting applications)

That notation suggestion helps, thanks. If I were to do the PR, I would also need a name for the manifold.
Something like Orthogonal-Columns-Matrices?

**Comment by vit-tucek** [comment]
This seems to be related to https://en.wikipedia.org/wiki/Generalized_flag_variety#Partial_flag_varieties
